Feature/event register form

alejandroguardadodev opened the pull request:
The events page has been created, which currently only contains the event registration form. Once we finish the event list page, we will update the links to read events and events/register.

Multiple components have been created, with highlights including one for the banner and various inputs and forms.

Multiple packages have been installed, which have aided me in form validation as well as responsive design handling.

alejandroguardadodev committed:
feat:event-page

- Implemented the events page, including the main event registration form.
- Developed multiple components for handling input.
- Added the "Banner" component to serve as a header for some pages.
- Installed multiple packages to enhance the responsiveness system; manage styles and forms more effectively.

alejandroguardadodev committed:
feat:event-page-forms

The "disabled" function has been added for inputs.
Checkboxes have been used to disable the inputs.
The input file bug has been fixed.
The function to reset the form has been added.

alejandroguardadodev committed:
feat:event-form-page-responsive

The responsive system has been enhanced with the grid system.
The events banner has been improved.
The events link has been added to both the main menu and the sidebar menu.

alejandroguardadodev committed:
feat:event-page

Fix reset form

alejandroguardadodev committed:
feat:event-list-page-first-commit

Implemented a new component for the events list. It's likely that once the design is approved, this component will replace the event cards currently displayed on the homepage.

Integrated a package to detect the device type (mobile, tablet, or desktop) and adapt the experience accordingly.

Updated the main menu in both mobile and desktop versions to enhance navigation and user experience.

Modified the banner design for a more appealing and effective appearance.

garcialuis commented on client/src/layouts/Events/RegisterEventForm/index.tsx:
```diff
@@ -0,0 +1,396 @@
+                <form className={styles.eventForm} onSubmit={methods.handleSubmit(onSubmit)}>
+                    <div className={styles.eventDetailsContainer}>
+                        <EventFormTitle title="Event Details" />
+                        <TextFieldBase id="name" title='Event name*' placeholder="Event name" fullwidth={true} />
+                        <TextFieldBase id="provider" title='Event provider*' placeholder="Event provider" fullwidth={true} />
```
I would prefer this as **Event Title** instead of **Event Name**
and **Organization Name** instead of **Event Provider**. 
but let me know what you think.

garcialuis commented on client/src/layouts/Events/RegisterEventForm/index.tsx:
```diff
@@ -0,0 +1,396 @@
+                        <TextFieldBase id="provider" title='Event provider*' placeholder="Event provider" fullwidth={true} />
+                        <div className={styles.inputGridContainer}>
+                            <div className={styles.gridFullColumn} style={{ gridColumn: 'span 6' }}>
+                                <TextFieldBase id="date" title='Event date*' placeholder="Event date" type="date" fullwidth={true} />
+                            </div>
+                            <div className={styles.gridHalfColumn} style={{ gridColumn: 'span 3' }}>
+                                <TextFieldBase id="starttime" disabled={wholeDayEvent} title='Start time' placeholder="HH:MM" type="time" fullwidth={true} />
+                            </div>
+                            <div className={styles.gridHalfColumn} style={{ gridColumn: 'span 3' }}>
+                                <TextFieldBase id="endtime" disabled={wholeDayEvent} title='End time' placeholder="HH:MM" type="time" fullwidth={true} />
```
We can have two sections - a "Start date" section in which they include the date and time that the event starts. And below that we can have an "End date" section for the date and time when the event ends. 

This would eliminate the need to have a "wholeDayEvent" checkbox and will also allow users to submit multi-day events such as conferences.

alejandroguardadodev committed:
feat:Register Event Form and Event list appareance

Add new form inputs
Change event not image preview
Fix bugs with whole day button
Fix Layout form

Event List page

alejandroguardadodev committed:
feat:Image Preview Err Msg

If there no image to upload, an error will show

alejandroguardadodev committed:
fix:Form Register Event

Added new fields
Added Event Hooks
Fix some validatation issued [StartTime, EndTime, EndDate]
Remove Event URL Inputs
Connect to DB
Get EVENTS and Categories

TODO:
Post: Create Event
Upcoming and Pasts Events

alejandroguardadodev commented:
I have finished to adapt the event register page. As well, I added the conection to the backend

alejandroguardadodev committed:
feat:update branch

alejandroguardadodev committed:
Merge branch feature/events-list

alejandroguardadodev committed:
fix:brach-err-update

There was an error at the moment to update the event-register-form branch so I have to update the whole visual elements

alejandroguardadodev committed:
feat:add-example-env

alejandroguardadodev committed:
fix:Event List Page

Add some "firewall" to always get an array

alejandroguardadodev committed:
fix:Event Page List - Pagination Err

Added some 'Firewall' to avoid any posible pagination err

alejandroguardadodev committed:
fix:Input-Select check empty array

alejandroguardadodev committed:
fix:Form Validation

Added Price (Any)
Change Price input type
Fix Min Date validation

alejandroguardadodev committed:
feat: removed image requirement

jipatino98 commented:
Good work Alex, I will merge into main branch
